src/components/JohnTheRipper/JohnTheRipper.tsx

```diff
@@ -71,32 +103,74 @@ const JohnTheRipper = () => {
     const onSubmit = async (values: FormValuesType) => {
         setLoading(true);
 
-        //if hash is not specified
-        if (values.hash.length === 0) {
-            const args = [`${values.filePath}`];
+        //if hash is stored in a textfile
```

Comment:
The onSubmit logic is Great. If you would like to improve on this, you can consider extracting the args into its own function that return value depend on the parameter input (functional approach). This would make the code a lot more scalable and less singleton.